hs_file_types/models/timeseries.py

```diff
@@ -733,6 +734,20 @@ def set_file_type(cls, resource, user, file_id=None, folder_path=''):
                 raise ValidationError(msg)
             return logical_file
 
+    def remove_aggregation(self):
+        """Deletes the aggregation object (logical file) *self* and the associated metadata
+        object. If the aggregation contains a system generated sqlite file that resource file also will be
+        deleted."""
+
+        # need to delete the system generated sqlite file
+        if self.has_csv_file:
+            # the sqlite file is a system generated file
+            for res_file in self.files.all():
+                if res_file.file_name.lower().endswith(".sqlite"):
+                    res_file.delete()
+                    break
+        super(TimeSeriesLogicalFile, self).remove_aggregation()
```

Comment:
Should the sqlite file deletion logic happen after line 749? This reasoning for this is that if there were any exception raised from line 749, the sqlite file would not be deleted which is what would be desired?

Reply:
If we call the super() (line 749) first we won't be able to know which sqlite file to delete since by that time the aggregation is deleted. That's why the sqlite file needs to be deleted first and then the aggregation.

Reply:
OK. That makes sense. Would it work then if you get a reference to res_file object for sqlite file before calling remove_aggregation() on its parent class to remove the aggregation, and delete this reference res_file object only after remove_aggregation() on its parent class finish without exception?

Reply:
Made the change as you suggested.